src/TestSuite/Constraint/EventFired.php

```diff
@@ -0,0 +1,50 @@
+<?php
+namespace Cake\TestSuite\Constraint;
+
+use Cake\Event\EventManager;
+use PHPUnit_Framework_Constraint;
+
+/**
+ * EventFired constraint
+ */
+class EventFired extends PHPUnit_Framework_Constraint
+{
+    /**
+     * Array of fired events
+     *
+     * @var EventManager
+     */
+    protected $_eventManager;
+
+    /**
+     * Constructor
+     *
+     * @param EventManager $eventManager Event manager to check
+     */
+    public function __construct($eventManager)
+    {
+        parent::__construct();
+        $this->_eventManager = $eventManager;
+    }
+
+    /**
+     * Checks if event is in fired array
+     *
+     * @param mixed $other Constraint check
+     * @return bool
+     */
+    public function matches($other)
+    {
+        return $this->_eventManager->getEventList()->hasEvent($other);
```

Comment:
Was waiting on some general yes/no before handling all cases haha. But yes, both constraints would check that we were tracking events and that there was an event list, if not it would throw an exception. (probably in __construct).